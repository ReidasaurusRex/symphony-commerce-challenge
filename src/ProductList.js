import React, { Component } from 'react';
import Product from './Product';
import $ from 'jquery';

class ProductList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      products: [],
      loading: true
    }
  }

  componentDidMount() {
    var url = "https://sneakpeeq-sites.s3.amazonaws.com/interviews/ce/feeds/store.js"
    $.ajax({
      url: url,
      type: 'GET',
      dataType: 'JSON',
      success: function(data) {
        this.setState({products: data.products, loading: false})
      }.bind(this),
      error: function(xhr, status, err) {
        console.log(url, status, err.toString());
      }
    })
  }

  render() {
    var items;
    if (this.state.loading) {
      items = (<li>Loading</li>);
    } else {
      items = this.state.products.map(p => {
        let imageHref = "http:" + p.mainImage.ref;
        return <Product key={p.id} name={p.name} price={p.defaultPriceInCents} image={imageHref} />
      });
    }
    console.log(items);
    return (
      <ul className="c-product-list">
        {items}
      </ul>
    );
  }
}

export default ProductList;