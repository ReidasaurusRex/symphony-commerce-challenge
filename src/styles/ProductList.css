.c-product-list {
  list-style-type: none;
}