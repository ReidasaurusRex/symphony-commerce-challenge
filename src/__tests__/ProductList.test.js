import React, { Component } from 'react';
import ProductList from "../ProductList";
import renderer from 'react-test-renderer';

test('Creates Product components for each each product', () => {
  const products = [
    {
      name: "Product 1",
      price: 100,
      image: "image-cdn.symphonycommerce.com/images/sites/fijiwater/1424736511908_6pack_straw_image_1.400w.jpg"
    }, {
      name: "Product 2",
      price: 200,
      image: "image-cdn.symphonycommerce.com/images/sites/fijiwater/1424736511908_6pack_straw_image_1.400w.jpg"
    }, {
      name: "Product 3",
      price: 300,
      image: "image-cdn.symphonycommerce.com/images/sites/fijiwater/1424736511908_6pack_straw_image_1.400w.jpg"
    }
  ];

  const productComponents = products.map(p => <Product name={p.name} price={p.price} image={p.image} />);

  const component = renderer.create(
    <ProductList>
      {productComponents}
    </ProductList>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});