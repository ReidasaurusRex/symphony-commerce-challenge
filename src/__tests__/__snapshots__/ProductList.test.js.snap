exports[`test ProductList creates Product components for each each product 1`] = `
<ul
  className="c-product-list">
  <li>
    Loading
  </li>
</ul>
`;
